src/main/java/org/cactoos/map/MapEnvelope.java

```diff
@@ -192,7 +199,10 @@ private Boolean contentsEqual(final Map<?, ?> other) {
                     final Y value = entry.getValue();
                     return new And(
                         () -> other.containsKey(key),
-                        () -> other.get(key).equals(value)
+                        () -> new EqualsNullable(
+                            other.get(key),
+                            value
+                        ).value()
```

Comment:
@Umbrah I think it would be even nicer if `EqualsNullable` was able to take `Scalar<Boolean>` in both parameters or one of either parameters so that you could write here:
```java
final X key = entry.getKey();
final Y value = entry.getValue();
return new And(
    () -> other.containsKey(key),
    new EqualsNullable(() -> other.get(key), value)
).value();
```

By the way, since calling `getKey` and `getValue` is costless, we can then write:
```java
(entry) -> new And(
    () -> other.containsKey(entry.getKey()),
    new EqualsNullable(() -> other.get(entry.getKey()), entry.getValue())
)
```